File: lib/core/models/place_prediction.dart
import 'package:flutter/foundation.dart';

@immutable
class PlacePrediction {
  const PlacePrediction({
    required this.placeId,
    required this.name,
    required this.address,
    this.distanceMeters,
  });

  final String placeId;
  final String name;
  final String address;

  /// Autocomplete(New) に origin を渡したときの現在地からの測地線距離（m, #146 C案）。
  /// 「近くの店」モードの距離昇順再ソートに使う。取得できない候補や origin 未指定では null。
  final int? distanceMeters;

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is PlacePrediction && placeId == other.placeId;

  @override
  int get hashCode => placeId.hashCode;
}

File: apps/web/src/places/place-prediction.ts
// 移植元: lib/core/models/place_prediction.dart

/// 地点検索（typeahead）の候補。座標は持たない——Google autocomplete が返さないため、
/// 確定時に `PlacesService.fetchLatLng` で引く2段フロー。
export interface PlacePrediction {
  readonly placeId: string;
  readonly name: string;
  readonly address: string;

  /// Autocomplete に origin を渡したときの現在地からの測地線距離（m, #146 C案）。
  /// 「近くの店」モードの距離昇順再ソートに使う。origin 未指定や取得できない候補では null。
  readonly distanceMeters: number | null;
}
